### PR TITLE
Enable remote logging and implement settings

### PR DESCRIPTION
**Pull request by matteosz**
This PR aims to:
- introduce the settings tab in the application; the settings have been designed to be persistent.
- give the user the possibility to activate the remote logging. During the POP party we are encountering issues like #1489 and very often we cannot either reproduce them or see the logs. So the idea is to enable an option in the settings which sends to a server all the logs.
- switched to Timber library for logging, as it can act as a centralized point of log collection and dispatch

UI Screen:

*As many files were modified to replace all the logs the code coverage is slightly under threshold*

### CODE REVIEW

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java [hunk 1]
```diff
@@ -185,6 +186,16 @@ public void setPageTitle(int titleId) {
     mPageTitle.postValue(titleId);
   }
 
+  public MutableLiveData<Boolean> isHome() {
+    return isHome;
+  }
+
+  public void setIsHome(boolean isHome) {
```
Comment (simone-kalbermatter) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java:193]: You could add a comment explaining what this does

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java [hunk 2]
```diff
@@ -135,13 +187,13 @@ private void handleWalletSettings() {
...
                     R.id.fragment_seed_wallet,
-                    SeedWalletFragment::new);
+                    SeedWalletFragment::newInstance);
               })
           .setNegativeButton(R.string.cancel, (dialog, which) -> dialog.dismiss())
           .show();
     } else {
       setCurrentFragment(
-          getSupportFragmentManager(), R.id.fragment_seed_wallet, SeedWalletFragment::new);
+          getSupportFragmentManager(), R.id.fragment_seed_wallet, SeedWalletFragment::newInstance);
```
Comment (simone-kalbermatter) [inline fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java:196]: What is the advantage of using newInstance here?

**Review by simone-kalbermatter — CHANGES_REQUESTED** [review]
Very impressive and clean work! Just a comment and a question.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java (hunk 2 above)
Comment (matteosz) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeActivity.java:196]: Actually none, it was just to be coherent with the rest of the code base

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java (hunk 1 above)
Comment (matteosz) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/home/HomeViewModel.java:193]: Sure 👍🏼

**Review by simone-kalbermatter — APPROVED** [review]
LGTM!

**Review by Warhedgehog — CHANGES_REQUESTED** [review]
Great work, this is will be a great improvement ! Also good idea with the architecture. A few minor things

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/LAONetworkManager.java [hunk 3]
```diff
@@ -100,20 +101,21 @@ private void processIncomingMessages() {
...
   public Completable catchup(Channel channel) {
-    Log.d(TAG, "sending a catchup to the channel " + channel);
+    Timber.tag(TAG).d("sending a catchup to the channel %s", channel);
     Catchup catchup = new Catchup(channel, requestCounter.incrementAndGet());
 
     return request(catchup)
         .map(ResultMessages.class::cast)
         .map(ResultMessages::getMessages)
-        .doOnError(error -> Log.d(TAG, "Error in catchup :" + error))
+        .doOnError(error -> Timber.tag(TAG).d(error, "Error in catchup"))
```
[the remark this answers is not in the document]
Comment (matteosz) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/remote/LAONetworkManager.java:115]: Yes, this is the syntax for Timber to signal a `Throwable`, so the first parameter has to be the error and the second one the message.
[Here](http://java.sun.com/javase/6/docs/api/java/lang/Throwable.html?is-external=true) is the doc about it.

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/security/elGamal/ElectionPublicKey.java [hunk 4]
```diff
@@ -106,9 +107,9 @@ public String encrypt(@NonNull byte[] message) {
       output.write(C.toBytes());
       result = output.toByteArray();
     } catch (IOException e) {
-      Log.d(
-          "Encryption: ",
-          "Something happened during the encryption, could concatenate the final result into a byte array");
+      Timber.tag(TAG)
+          .d(
+              "Something happened during the encryption, could concatenate the final result into a byte array");
```
[the remark this answers is not in the document]
Comment (matteosz) [reply fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/security/elGamal/ElectionPublicKey.java:112]: Yeah right, I missed it

**Review by Warhedgehog — APPROVED** [review]
LGTM

**Pull request merged**